pkg/kubectl/cmd/apply.go

```diff
@@ -245,7 +245,13 @@ func (p *patcher) patchSimple(obj runtime.Object, modified []byte, source, names
 	// Create the versioned struct from the type defined in the restmapping
 	// (which is the API version we'll be submitting the patch to)
 	versionedObject, err := api.Scheme.New(p.mapping.GroupVersionKind)
-	if err != nil {
+	switch {
+	case runtime.IsNotRegisteredError(err):
+		versionedObject, _, err = p.decoder.Decode(current, nil, nil)
```

Comment:
the fact that we can't construct `api.Scheme.New` indicates we cannot get a versioned object struct. I think we should instead switch to generic patch computation or complete replace mode